File: gcj2014QRA.py
#!/usr/bin/env python
import sys

def solve():
    n1 = int(sys.stdin.readline())
    a1 = [map(int,sys.stdin.readline().split()) for _ in range(4)]
    n2 = int(sys.stdin.readline())
    a2 = [map(int,sys.stdin.readline().split()) for _ in range(4)]
    u = set(a1[n1-1]).intersection(set(a2[n2-1]))
    if len(u) < 1:
        return "Volunteer cheated!"
    if len(u) > 1:
        return "Bad magician!"
    return list(u)[0]

for x in range(int(sys.stdin.readline())):
    print("Case #"+str(x+1)+": "+str(solve()))

File: gcj2014QRA.swift
#!/usr/bin/env swift
import Foundation
func getContens() -> String {
    return NSString(data: NSFileHandle.fileHandleWithStandardInput().availableData, encoding:NSUTF8StringEncoding)! as String
}
func strip(s: String, charset: NSCharacterSet) -> String {
    return s.stringByTrimmingCharactersInSet(charset)
}
func strip(s: String) -> String {
    return strip(s,NSCharacterSet.whitespaceAndNewlineCharacterSet())
}
func lines(s: String) -> [String] {
    return strip(s,NSCharacterSet.newlineCharacterSet()).componentsSeparatedByCharactersInSet(NSCharacterSet.newlineCharacterSet())
}
func getInts(s: String) -> [Int] {
    return split(s){$0==" "}.map({$0.toInt()!})
}
func solve(input: ArraySlice<String>) -> String {
    let an = getInts(input[input[0].toInt()!+0])
    let bn = getInts(input[input[5].toInt()!+5])
    let s = Set(an).intersect(bn)
    switch s.count {
        case 0: return "Volunteer cheated!"
        case 1: return Array(s)[0].description
        default: return "Bad magician!"
    }
}
let cs = lines(getContens())
for t in 1...cs[0].toInt()! {
    let input = cs[1+(t-1)*10..<1+t*10]
    println("Case #\(t): \(solve(input))")
}
